src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -607,13 +607,23 @@ private PollingResult compareRemoteRevisionWithImpl(Job<?, ?> project, Launcher
             listener.getLogger().println("[poll] Last Built Revision: " + buildData.lastBuild.revision);
         }
 
-        final EnvVars pollEnv = project instanceof AbstractProject ? GitUtils.getPollEnvironment((AbstractProject) project, workspace, launcher, listener, false) : lastBuild.getEnvironment(listener);
+        EnvVars pollEnv;
+        if (project instanceof AbstractProject) {
+            pollEnv = GitUtils.getPollEnvironment((AbstractProject) project, workspace, launcher, listener, false);
```

Comment:
For me it seems to be a fragile hack in the original code, because it also leads to passing of multiple other environment variables to the polling logic. May cause some funny results if the build uses EnvInject plugin to pass variables to Git CLI. WDYT @MarkEWaite?